src/lib/swapper/api.ts

```diff
@@ -288,15 +288,21 @@ export type GetUnsignedTxArgs = {
   feeAssetUsdRate: string
 } & FromOrXpub
 
+// the client should never need to know anything about this payload, and since it varies from
+// swapper to swapper, the type is declared this way to prevent generics hell while ensuring the
+// data originates from the correct place (assuming no casting).
+export type UnsignedTx2 = Nominal<object, 'UnsignedTx2'>
```

Comment:
this change is key to allow all swappers to conform to the same interface with entering generics hell. Feel pretty strongly about keeping this as an opaque type instead of a generic interface

Reply:
Can we make this a `Record<string, any>` instead of an object?
The `object` type in TS is an old construct from [TS 2.2](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#object-type) which represents anything that's not a primitive (i.e an "object" at the very low "everything is an object" level, e.g a function, an array...)

Reply:
4e8a1196a9684a0524b90a32228ce31e9cd66962